File: app/src/androidTest/java/AdapterClass.java

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.accounttcreation.DataClass;
import com.example.accounttcreation.R;

import org.junit.Test;

import java.text.CollationElementIterator;
import java.util.ArrayList;

import javax.annotation.Nonnull;

public class AdapterClass extends RecyclerView.Adapter<AdapterClass.ViewHolder> {

    ArrayList<DataClass> list;
    public AdapterClass (ArrayList<DataClass> list)
    {
        this.list = list;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.items_db_recycler,viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        viewHolder.email.setText(list.get(i).getEmail());
        viewHolder.faculty.setText(list.get(i).getFaculty());
        viewHolder.name.setText(list.get(i).getName());
        viewHolder.title.setText(list.get(i).getTitle());

    }

    @Override
    public int getItemCount() {

        return list.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        TextView email, faculty, name, title;
        public ViewHolder(@Nonnull View ListView){

            super(ListView);
            email = ListView.findViewById(R.id.EmailDB);
            faculty = ListView.findViewById(R.id.FacultyDB);
            name = ListView.findViewById(R.id.NameDB);
            title = ListView.findViewById(R.id.TitleDB);
        }

    }
}

File: app/src/main/java/com/example/accounttcreation/AdapterClass.kt
package com.example.accounttcreation

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class AdapterClass(private val userList : ArrayList<DataClass>)  : RecyclerView.Adapter<AdapterClass.ViewHolder>(){


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.items_db_recycler, parent, false)
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val currentItem = userList[position]
        holder.Email.text = currentItem.Email
        holder.Faculty.text = currentItem.Faculty
        holder.Name.text = currentItem.Name
        holder.Title.text = currentItem.Title
    }

    override fun getItemCount(): Int {
        return userList.size
    }


    class ViewHolder(itemView : View) : RecyclerView.ViewHolder(itemView){

        val Email : TextView  = itemView.findViewById(R.id.EmailDB)
        val Faculty : TextView = itemView.findViewById(R.id.FacultyDB)
        val Name : TextView = itemView.findViewById(R.id.NameDB)
        val Title : TextView= itemView.findViewById(R.id.TitleDB)


    }
}